### ISSUE TITLE
When transcoding with subtitles (burning in), only the default audio track is available.

### ISSUE THREAD

**Issue by nothing2obvi** [jellyfin-archive/jellyfin-roku-legacy#1179]
## Software Versions

- Jellyfin Server Version: 10.8.9
- Roku Client Version: 1.6.4

## Describe the bug
Most of my anime has English as the default track with Japanese as the secondary track. Also, most of the subtitles on my anime are PGSSUB, DVDSUB, or ASS, so I believe they need to be transcoded (burned in). I want to use the secondary audio track, Japanese, with subtitles.

**Scenario 1**
When selecting a subtitle track by pressing down on the Roku remote, I am able to get subtitles to burn in. However, when I then pick the audio track, the only choice there is “Default,” which in my case is English.

**Scenario 2**
Now, if I _first_ choose the secondary audio track, Japanese, I can get it to play. And when I then select subtitles by pressing down on the Roku remote, the subtitles get burned in. However, the audio track then reverts back to the “Default” audio track, English. Also, it is the only choice available for audio track.

## How To Reproduce

**Scenario 1**
1. Choose anime that needs subtitles transcoding
2. Play anime on Roku
3. Select subtitles on Roku by pressing down on the Roku remote; it works
4. Go to Audio track and find that “Default” is the only one available

**Scenario 2**
1. Choose anime that needs subtitles transcoding
2. Play anime on Roku
3. Select secondary audio track; it works
4. Select subtitles on Roku by pressing down on the Roku remote; it works
5. Find that the audio track has reverted back to the “Default” track
6. Go to Audio track and find that “Default” is the only one available

## Expected behavior

**Scenario 1**
After selecting subtitles to burn in, all audio tracks should be available to choose from.

**Scenario 2**
After selecting the secondary audio track and then choosing subtitles to burn in, the audio track should _not_ revert back to the default audio track. Also, all audio tracks should be available to choose from.

## Connection Information

- Is server local or remote? Remote
- Is server connection HTTP or HTTPS? HTTPS

## Additional Context
This may be related to #923, #657, and https://github.com/jellyfin/jellyfin/issues/7896.

**Comment by cewert** [comment]
Hello and thank you for the bug report!

In the Jellyfin web client user settings under Playback -> Audio, can you please try turning "Play default audio track regardless of language" off and see if that fixes your problem?

![user-settings](https://user-images.githubusercontent.com/16514652/230907053-ba210eb5-b4fb-4720-9532-c52bcfb3780d.PNG)

![user-settings-default-audio](https://user-images.githubusercontent.com/16514652/230907060-5c0749df-6b4c-42a4-9940-c82e2dfea0ea.PNG)

**Comment by nothing2obvi** [comment]
Hi @cewert, I just tried what you suggested for the specific user logged in on Jellyfin-Roku. The problem persists.

I know a workaround to this might be to switch the audio tracks on all my files, but since I don’t have this issue on Jellyfin-Kodi (though I haven’t tested this on Jellyfin-AndroidTV yet), I’d like to see if I can avoid doing that.

Thanks for your help!

**Comment by cewert** [comment]
> I’d like to see if I can avoid doing that.

Totally understandable. Thanks for trying!

I'm starting to think this is the same issue as #923 which seems to be caused because the files are being played directly and not transcoded. Couple questions:

- The files you are having problems with, are they direct playing or transcoding? you can check this on the server dashboard and also the server logs
- If you go into your Jellyfin server settings and set `Internet streaming bitrate limit` to a very low number (either globally in dashboard -> playback -> streaming or by user in dashboard->users) to force the files to transcode does that fix your issue?

**Comment by nothing2obvi** [comment]
Hey @cewert, I think I found the solution to this issue, but first I’ll answer your questions.

1.) The files are transcoding. The reason on the dashboard is that the subtitle codec is not supported. If I play the files without the subtitles, playing either audio track, it is direct-playing.

2.) I tried setting ‘Internet streaming bitrate limit’ to a very low value, and the issue persisted.

However, in my original post, when I said “Go to audio track,” I didn’t specify that the way I did that was by pressing the `*` button while the episode is playing, which opens “Accessibility and captions,” and trying to select audio track from there. So far, by this method, while burning in subtitles I still cannot change the audio track from “Default,” English.

_But_ I just tried this: in the _episode list,_ I pressed `*` on the episode and it lists the two audio tracks. I selected the second audio track, Japanese, pressed back, then pressed play on the episode. The episode plays in Japanese. I then pressed down on the Roku remote to select subtitles, they burned in, and the audio track _did not_ revert back to the default, English! If I now press `*` and go to “Accessibility and captions,” it says that _this audio track,_ Japanese, is the “Default” and is the only option to choose from.

The only problem with this method is that once the episode ends, I must do the same routine for the next episode by going back to the episode list and pressing `*` to select the audio track again. But honestly, this is fine for me.

**TL;DR:** To get around this issue, _first_ select the audio track from the _episode list_ by pressing `*` then choose subtitles while the episode is playing by pressing down.

I will close this issue, but feel free to reopen if you think it needs to be.

Thanks again for your help!

**Comment by cewert** [comment]
That's awesome you got it working and thank you for reporting back!

The star menu that pops up while a video is playing is a Roku menu and not a Jellyfin menu so that might explain why it wasn't working. I don't believe we have any control over the star button during video but I could be wrong. As long as everything works when using features coded by the Jellyfin app I think it makes sense to close this.

**Comment by nothing2obvi** [comment]
For sure!

Thanks again for all your help.
